state/postgres_database.go

```diff
@@ -21,4 +21,33 @@ type PostgresDatabase struct {
 	// This is used to track whether the database needs to be vacuumed in order to prevent multixact ID wraparound or to
 	// allow pg_multixact to be shrunk. It is the minimum of the per-table pg_class.relminmxid values.
 	MinimumMultixactXID Xid
+
+	FrozenXIDAge uint32 // Age of FrozenXID
+	MinMXIDAge   uint32 // Age of MinimumMultixactXID
+
+	XactCommit   int64 // Number of transactions in this database that have been committed
+	XactRollback int64 // Number of transactions in this database that have been rolled back
+}
+
+// PostgresDatabaseStats - Database statistics for a single database
+type PostgresDatabaseStats struct {
+	TransactionCount uint64 // Transaction count of this database
+}
+
+// PostgresDatabaseStatsMap - Map of database statistics (key = database Oid)
+type PostgresDatabaseStatsMap map[Oid]PostgresDatabaseStats
+
+// DiffedPostgresDatabaseStat - Database statistics for a single database as a diff
+type DiffedPostgresDatabaseStats struct {
+	TransactionsPerSecond uint64
```

Comment:
I think we should also diff `XactCommit` and `XactRollback` the same way we do for TPS.

Reply:
Ah wait, I think I'm confused. So you wanted to have the diff, not the total count? If so, maybe it's better to clarify it in the naming and/or comment.

Reply:
Oh, actually, looking the code more, looks like others are not using the diff value also, but passing the absolute value and do the whole calculation by using the `snapshot.collected_interval_secs` + getting the last snapshot 🤔  Maybe you meant this?
I'll hold on tweaking this until the further clarification.

Reply:
Generally, the way it works today for most statistics, such as table statistics ("tuples updated", etc):

* First full snapshot:
  * We collect data point A
  * We don't send any data, since there is nothing to diff against
  * On the server side we don't see any data yet (since the diff was not done)
* Second full snapshot (~600 seconds later)
  * We collect data point B
  * We calculate "B - A" and send this as the diff to the server (this represents the activity that happened since the last full snapshot)
  * On the server side, we get the diff and then divide it by the time since the last snapshot (~600 seconds) to get a per second value

We do not pass in the raw values of B or A, since they don't really tell us anything - only the diff between the two values is meaningful, since it represents the amount of activity within a known time interval (i.e. between two full snapshots).

There is two things that I wanted to comment on in the context of this PR:
- I think it would be good to know "how many commits happened" separately from "how many rollbacks happened" (since it allows us to graph that in the future, even if for now we mostly care about the sum of both to understand TXID consumption) -- your initial code here sent the raw values (i.e. A or B), not the diff between two snapshots, i.e. B - A)
- To align with the other collector statistics code (with the exception of system statistics), I think we should not make the calculation of "transactions per second" on the collector side, but on the server side - and instead only send the difference in how many transactions occurred between snapshots

Does that clarify things? (happy to chat more on Slack/etc)

Reply:
Ohhh thanks for explaining! I didn't understand this part at all (TIL schema stats are holding diff info, not the total count)!
I think that's also why I'm missing the change to the `PostgresRelationStats` in this PR, the table level stats change should have been made there, not `PostgresRelation` (well technically `frozen_xid_age` doesn't need to be stored in `PersistedState` so it could still be there, but looks like we're doing it anyways with the stats like `NLiveTup` so we can put in `PostgresRelationStats`).

> To align with the other collector statistics code (with the exception of system statistics)

by now, I think it's clear that I was copy&pasting from the wrong place 🙃